components/channel_notifications_modal/components/notification_section.tsx

```diff
@@ -1,82 +1,38 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import PropTypes from 'prop-types';
 import React from 'react';
 
 import {NotificationSections, NotificationLevels} from 'utils/constants';
 
 import CollapseView from './collapse_view';
 import ExpandView from './expand_view';
 
-export default class NotificationSection extends React.PureComponent {
-    static propTypes = {
-
-        /**
-         * Notification section
-         */
-        section: PropTypes.string.isRequired,
-
-        /**
-         * Expand if true, else collapse the section
-         */
-        expand: PropTypes.bool.isRequired,
-
-        /**
-         * Member's desktop notification level
-         */
-        memberNotificationLevel: PropTypes.string.isRequired,
-
-        /**
-         * Member's desktop_threads notification level
-         */
-        memberThreadsNotificationLevel: PropTypes.string,
-
-        /**
-         * Ignore channel-wide mentions @channel, @here and @all
-         */
-        ignoreChannelMentions: PropTypes.string,
-
-        /**
-         * User's global notification level
-         */
-        globalNotificationLevel: PropTypes.string,
-
-        /**
-         * onChange handles update of desktop notification level
-         */
-        onChange: PropTypes.func.isRequired,
-
-        /**
-         * onChangeThreads handles update of desktop_threads notification level
-         */
-        onChangeThreads: PropTypes.func,
-
-        /**
-         * Submit function to save notification level
-         */
-        onSubmit: PropTypes.func.isRequired,
-
-        /**
-         * Update function to to expand or collapse a section
-         */
-        onUpdateSection: PropTypes.func.isRequired,
-
-        /**
-         * Error string from the server
-         */
-        serverError: PropTypes.string,
-    }
-
-    handleOnChange = (e) => {
+type NotificationSectionProps = {
+    section: string,
+    expand: boolean,
+    memberNotificationLevel: string,
+    memberThreadsNotificationLevel: string,
+    ignoreChannelMentions: string,
+    globalNotificationLevel: string,
+    onChange: (value: string) => void,
+    onChangeThreads: (value: string) => void,
```

Comment:
`onChangeThreads` is optional

Reply:
to turn this function into non-obligatory, we need to put a second value when it is not used, since the 'undefined' is not accepted in TypeScript, we put the type 'any'.
Contribution made by my partner Poliana Queiroz